File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/reactiveobject/ReadHistoryEventResponse.java
package org.opcfoundation.webserver.service.message.reactiveobject;

import org.eclipse.milo.opcua.stack.core.types.builtin.DataValue;
import org.eclipse.milo.opcua.stack.core.types.structured.HistoryEventFieldList;

import java.util.List;

public class ReadHistoryEventResponse {
    private List<HistoryEventFieldList> events;
    private Boolean containsMoreData;

    public ReadHistoryEventResponse(List<HistoryEventFieldList> events, Boolean containsMoreData)
    {
        this.events = events;
        this.containsMoreData = containsMoreData;
    }

    public List<HistoryEventFieldList> getEvents() {
        return events;
    }

    public Boolean getContainsMoreData() {
        return containsMoreData;
    }
}

File: opcua-webservice-node/src/server/service/message/reactiveobject/ReadHistoryEventResponse.ts
import { UaHistoryEventFieldList } from "opcua-webapi-ts";

export class ReadHistoryEventResponse {
    private readonly _events: Array<UaHistoryEventFieldList>;
    private readonly _containsMoreData: boolean;

    constructor(events: Array<UaHistoryEventFieldList>, containsMoreData: boolean) {
        this._events = events;
        this._containsMoreData = containsMoreData;
    }

    get events(): Array<UaHistoryEventFieldList> {
        return this._events;
    }

    get containsMoreData(): boolean {
        return this._containsMoreData;
    }
}